conda/install.py

```diff
@@ -635,8 +642,19 @@ def messages(prefix):
     finally:
         rm_rf(path)
 
+
+def ensure_write(prefix, dist):
+    if not can_open_all_files_in_prefix(prefix, dist['files']):
```

Comment:
Well shoot -- `dist` in this case is a string.  The `can_open_all_files_in_prefix` assumes that there's already associated meta data.  Refactoring this code.